src/masternodeman.cpp

```diff
@@ -1467,32 +1465,30 @@ void CMasternodeMan::CheckMasternode(const CPubKey& pubKeyMasternode, bool fForc
     }
 }
 
-bool CMasternodeMan::IsMasternodePingedWithin(const CTxIn& vin, int nSeconds, int64_t nTimeToCheckAt)
+bool CMasternodeMan::IsMasternodePingedWithin(const COutPoint& outpoint, int nSeconds, int64_t nTimeToCheckAt)
 {
     LOCK(cs);
-    CMasternode* pMN = Find(vin);
-    if(!pMN) {
-        return false;
-    }
-    return pMN->IsPingedWithin(nSeconds, nTimeToCheckAt);
+    CMasternode* pmn = Find(outpoint);
+    return pmn ? pmn->IsPingedWithin(nSeconds, nTimeToCheckAt) : false;
 }
 
-void CMasternodeMan::SetMasternodeLastPing(const CTxIn& vin, const CMasternodePing& mnp)
+void CMasternodeMan::SetMasternodeLastPing(const COutPoint& outpoint, const CMasternodePing& mnp)
 {
     LOCK(cs);
-    CMasternode* pMN = Find(vin);
-    if(!pMN)  {
+    CMasternode* pmn = Find(outpoint);
+    if(!pmn) {
         return;
     }
-    pMN->lastPing = mnp;
+    pmn->lastPing = mnp;
```

Comment:
Hmmm... seems like critical session isn't doing much in CMasternode at all... but it doesn't break anything either, so I'd keep it for now. I have another set of changes for locks specifically (in local repo), so that probably would be a better place to change this.